### PR TITLE
Added ribbon tests

### PR DESCRIPTION
**Pull request by juangon**

### CODE REVIEW

File: netflix/ribbon/pom.xml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+            <groupId>org.wildfly.swarm</groupId>
+            <artifactId>jaxrs</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.wildfly.swarm</groupId>
+            <artifactId>ribbon</artifactId>
+        </dependency>
+        <dependency>
+          <groupId>org.wildfly.swarm</groupId>
+          <artifactId>topology-jgroups</artifactId>
```
Comment (Ladicek) [inline netflix/ribbon/pom.xml]: Look into the `topology` or `topology-webapp` tests for a mock topology implementation.

File: netflix/ribbon/src/main/java/org/wildfly/swarm/ts/netflix/ribbon/HelloResource.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+        Topology topology;
+        try {
+            topology = Topology.lookup();
+            Map<String, List<Topology.Entry>> entries = topology.asMap();
+            for (String key : entries.keySet()) {               
+                response = response + key;
+            }
+        } catch (Exception e) {
+            e.printStackTrace();
```
Comment (Ladicek) [inline netflix/ribbon/src/main/java/org/wildfly/swarm/ts/netflix/ribbon/HelloResource.java]: Please don't.

File: netflix/ribbon/src/test/java/org/wildfly/swarm/ts/netflix/ribbon/RibbonTestFallbackHandler.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.UnpooledByteBufAllocator;
+import rx.Observable;
+
+public class RibbonTestFallbackHandler implements FallbackHandler<ByteBuf>{
+
+    @Override
+    public Observable<ByteBuf> getFallback(HystrixInvokableInfo<?> hystrixInfo, Map<String, Object> requestProperties) {
+        String fallback = "Fallback string";
+        byte[] bytes = fallback.getBytes();
```
Comment (Ladicek) [inline netflix/ribbon/src/test/java/org/wildfly/swarm/ts/netflix/ribbon/RibbonTestFallbackHandler.java]: Please specify a charset. `StandardCharsets.UTF_8` should work.

File: netflix/ribbon/src/test/java/org/wildfly/swarm/ts/netflix/ribbon/RibbonTopologyTest.java [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+                    .withUriTemplate("http://localhost:8080/")
+                    .withMethod("GET")
+                    .withFallbackProvider(new RibbonTestFallbackHandler())
+                    .build();
+            RibbonRequest<ByteBuf> ribbonRequest = jaxrsTemplate.requestBuilder().build();
+
+            Observable<ByteBuf> observable = ribbonRequest.observe();
+            ByteBuf byteBuf = observable.toBlocking().single();
+
+            assertThat(byteBuf.toString(Charset.defaultCharset())).isEqualTo("ts-netflix-ribbon");
```
Comment (Ladicek) [inline netflix/ribbon/src/test/java/org/wildfly/swarm/ts/netflix/ribbon/RibbonTopologyTest.java]: `StandardCharsets.UTF_8` please.

File: netflix/ribbon/src/test/java/org/wildfly/swarm/ts/netflix/ribbon/RibbonTopologyTest.java [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+
+            swarm.stop();
+            observable = ribbonRequest.observe();
+            assertThat(observable.toBlocking().single().toString(Charset.defaultCharset())).isEqualTo("Fallback string");
```
Comment (Ladicek) [inline netflix/ribbon/src/test/java/org/wildfly/swarm/ts/netflix/ribbon/RibbonTopologyTest.java]: Same as above.

**Comment by juangon** [comment]
@Ladicek pushed suggested changes. I didn't see this mock topology thing. Thanks for that!

**Comment by Ladicek** [comment]
Thanks!

**Pull request merged**